RiotSwiftUI/Modules/Settings/ChangePassword/Coordinator/ChangePasswordCoordinator.swift

```diff
@@ -0,0 +1,134 @@
+    
+    private var indicatorPresenter: UserIndicatorTypePresenterProtocol
+    private var loadingIndicator: UserIndicator?
+    
+    private var currentTask: Task<Void, Error>? {
+        willSet {
+            currentTask?.cancel()
+        }
+    }
+    
+    // MARK: Public
+
+    // Must be used only internally
+    var childCoordinators: [Coordinator] = []
+    var callback: (@MainActor () -> Void)?
+    
+    // MARK: - Setup
+    
+    @MainActor init(parameters: ChangePasswordCoordinatorParameters) {
+        self.parameters = parameters
+        
+        let viewModel = ChangePasswordViewModel()
+        let view = ChangePasswordScreen(viewModel: viewModel.context)
+        changePasswordViewModel = viewModel
+        changePasswordHostingController = VectorHostingController(rootView: view)
+        changePasswordHostingController.vc_removeBackTitle()
+        changePasswordHostingController.enableNavigationBarScrollEdgeAppearance = true
+        
+        indicatorPresenter = UserIndicatorTypePresenter(presentingViewController: changePasswordHostingController)
+    }
+    
+    // MARK: - Public
+    
+    func start() {
+        MXLog.debug("[ChangePasswordCoordinator] did start.")
+        Task { await setupViewModel() }
+    }
+    
+    func toPresentable() -> UIViewController {
+        return self.changePasswordHostingController
+    }
+    
+    // MARK: - Private
+    
+    /// Set up the view model. This method is extracted from `start()` so it can run on the `MainActor`.
+    @MainActor private func setupViewModel() {
+        changePasswordViewModel.callback = { [weak self] result in
+            guard let self = self else { return }
+            MXLog.debug("[ChangePasswordCoordinator] ChangePasswordViewModel did complete with result: \(result).")
+            
+            switch result {
+            case .submit(let oldPassword, let newPassword, let signoutAllDevices):
+                self.changePassword(from: oldPassword, to: newPassword, signoutAllDevices: signoutAllDevices)
+            }
+        }
+    }
+    
+    /// Show an activity indicator whilst loading.
+    @MainActor private func startLoading() {
+        loadingIndicator = indicatorPresenter.present(.loading(label: VectorL10n.loading, isInteractionBlocking: true))
+    }
+    
+    /// Hide the currently displayed activity indicator.
+    @MainActor private func stopLoading() {
+        loadingIndicator = nil
+    }
+    
+    /// Submits a reset password request with signing out of all devices option
+    @MainActor private func changePassword(from oldPassword: String, to newPassword: String, signoutAllDevices: Bool) {
+        startLoading()
+
+        currentTask = Task { [weak self] in
+            do {
+                try await parameters.restClient.changePassword(from: oldPassword, to: newPassword, logoutDevices: signoutAllDevices)
+
+                // Shouldn't be reachable but just in case, continue the flow.
```

Comment:
I don't think this comment is relevant for this API call as it should succeed. You can probably remove the `catch is CancellationError` from here too.